numpy/lib/tests/test_nanfunctions.py

```diff
@@ -388,6 +410,101 @@ def test_empty(self):
             assert_equal(res, tgt)
 
 
+class TestNanFunctions_CumSumProd(TestCase, SharedNanFunctionsTestsMixin):
+
+    nanfuncs = [np.nancumsum, np.nancumprod]
+    stdfuncs = [np.cumsum, np.cumprod]
+
+    def test_allnans(self):
+        # Check for FutureWarning
+        for f, tgt_value in zip(self.nanfuncs, [0, 1]):
+            with warnings.catch_warnings(record=True) as w:
```

Comment:
Yeah, sure, nothing wrong with this here. Just thought it was a bit
funny and was wondering if someone disagrees ;). I probably should make
these context managers use my context manager instead of catch
warnings, if we merge it. Though I think I would do that after the
other is in.